alica_engine/src/engine/model/Synchronisation.cpp

```diff
@@ -1,19 +1,27 @@
 #include "engine/model/Synchronisation.h"
 #include "engine/model/Plan.h"
 #include "engine/model/Transition.h"
+#include "engine/modelmanagement/factories/Factory.h"
+#include <sstream>
 
 namespace alica
 {
 
-Synchronisation::Synchronisation()
-        : _failOnSyncTimeout(false)
-        , _syncTimeout(AlicaTime::milliseconds(3000))
+Synchronisation::Synchronisation(const YAML::Node& synchronisationNode, Plan* plan)
+        : _plan(plan)
         , _talkTimeout(AlicaTime::milliseconds(30))
-        , _plan(nullptr)
+        , _syncTimeout(AlicaTime::milliseconds(3000))
+        , _failOnSyncTimeout(false)
 {
+    _failOnSyncTimeout = Factory::getValue<bool>(synchronisationNode, alica::Strings::failOnSyncTimeout);
+    _syncTimeout = AlicaTime::milliseconds(Factory::getValue<int>(synchronisationNode, alica::Strings::syncTimeout));
+    _talkTimeout = AlicaTime::milliseconds(Factory::getValue<int>(synchronisationNode, alica::Strings::talkTimeout));
```

Comment:
Should not be using Factory here.
In general I think we should try to parse the yaml node before construction, even if it means we have an extra struct for each of these classes giving the (parsed) input params